arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/Hub.java

```diff
@@ -652,7 +652,7 @@ public int contextNumberNew(WorldView world) {
     if (pch.exceptions().any()
         || opCode.getData().instructionFamily().equals(InstructionFamily.HALT)
         || opCode.getData().instructionFamily().equals(InstructionFamily.INVALID)) {
-      return this.callStack().caller().contextNumber();
+      return this.callStack.depth() == 0 ? 0 : this.callStack().caller().contextNumber(); //TODO: when callStackDepth = 0
```

Comment:
In the arithmetization we have a shared column called CALLER_CN (if I'm not misremembering the name) which is context-constant etc. In any case, this number will always contain the proper parent execution context. In case you are the root context you really have 2 parent context numbers:
- CALLER_CONTEXT_NUMBER (shared column, the one I mentioned above)
- context/CALL_DATA_CONTEXT_NUMBER (column in the context perspective, which points to the special execution context number whose RAM contains the transaction call data.)